spec.bs

```diff
@@ -0,0 +1,393 @@
+
+Issue: New enum needed for bid reject reasons.
+
+Storage {#storage}
+==================
+
+A user agent holds an <dfn>aggregatable report cache</dfn>, which is an [=ordered set=] of [=aggregatable reports=].
+
+Vendor-specific values {#vendor-specific-values}
+================================================
+
+Issue: Should we use the term "implementation-defined" instead?
+
+<dfn>Max stored reports</dfn> is a positive integer that controls how many unsent reports can be queued waiting.
+
+Issue: More
+
+Algorithms {#algorithms}
+====================
+
+
+<!-- A {{WorkletGlobalScope}} |scope|'s <dfn algorithm for="PrivateAggregation">api identifier</dfn> is the result of:
+1. If |scope|'s [=worklet global scope type=] is {{FledgeWorkletGlobalScope}}, then return "<code>[=worklet type identifier/fledge=]</code>".
+1. [=Assert=]: |scope|'s [=worklet global scope type=] is {{SharedStorageWorkletGlobalScope}}.
+1. Return "<code>[=worklet type identifier/shared-storage=]</code>". -->
+
+To <dfn algorithm for="PrivateAggregation">obtain a reporting endpoint</dfn> given an [=Origin=] |reportingOrigin| and [=worklet type identifier=] |api|:
+1. [=Assert=]: |reportingOrigin| is a [=potentially trustworthy origin=].
+1. Let |path| be the [=string/concatenation=] of «"<code>.well-known/private-aggregation/report-</code>", |api|».
+1. Let |base| be the result on running the [=URL parser=] on the [=serialization of an origin|serialization=] of |reportingOrigin|.
+1. Return the result of running the [=URL parser=] on |path| with |base|.
+
+To <dfn for="PrivateAggregation">obtain an aggregatable report</dfn> given an [=Origin=] |reportingOrigin|, a [=worklet type identifier=] |api| and a [=ordered set=] of {{PAHistogramContribution}} |contributions|:
+1. [=Assert=]: |reportingOrigin| is a [=potentially trustworthy origin=].
+1. Let |reportTime| be the result of running [=PrivateAggregation/obtain a report delivery time=].
+1. Let |report| be a new [=aggregatable report=] with the items:
+    : [=aggregatable report/reporting endpoint=]
+    :: The result of [=PrivateAggregation/obtaining a reporting endpoint=] given |reportingOrigin| and |api|.
+    : [=aggregatable report/original report time=]
+    :: |reportTime|
+    : [=aggregatable report/report time=]
+    :: |reportTime|
+    : [=aggregatable report/contributions=]
+    :: |contributions|
+    : [=aggregatable report/api=]
+    :: |api|
+    : [=aggregatable report/report ID=]
+    :: The result of [=generating a random UUID=].
+
+Scheduling reports {#scheduling-reports}
+----------------------------------------
+
+To <dfn algorithm for="PrivateAggregation">obtain a report delivery time</dfn>, run the following steps:
+
+Issue: Fill in
+
+The <dfn algorithm for="PrivateAggregation">report scheduling steps</dfn> given an [=Origin=] |reportingOrigin|, a [=worklet type identifier=] |api| and a [=ordered set=] of {{PAHistogramContribution}} |contributions| are as follows:
```

Comment:
```suggestion
The <dfn algorithm for="PrivateAggregation">report scheduling steps</dfn> given an [=Origin=] |reportingOrigin|, a [=worklet type identifier=] |api| and an [=ordered set=] of {{PAHistogramContribution}} |contributions| are as follows:
```

Reply:
Switched to a list actually -- doesn't really make sense for it to be a set